src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -254,29 +253,54 @@ protected async Task<(Guid, string projectName, SourceText)> GetGuidAndProjectNa
 
                 var projectName = document.Project.Name;
-                var guid = workspace.GetProjectGuid(document.Project.Id);
+                var guid = vsWorkspace?.GetProjectGuid(document.Project.Id) ?? Guid.Empty;
 
                 var sourceText = await document.GetTextAsync(CancellationToken).ConfigureAwait(false);
                 return (guid, projectName, sourceText);
             }
 
-            protected async Task<Entry> CreateDocumentSpanEntryAsync(
+            protected async Task<Entry> TryCreateDocumentSpanEntryAsync(
                 RoslynDefinitionBucket definitionBucket,
                 DocumentSpan documentSpan,
                 HighlightSpanKind spanKind)
             {
                 var document = documentSpan.Document;
                 var (guid, projectName, sourceText) = await GetGuidAndProjectNameAndSourceTextAsync(document).ConfigureAwait(false);
+                var excerptResult = await ExcerptAsync(sourceText, documentSpan).ConfigureAwait(false);
 
-                var classifiedSpansAndHighlightSpan =
-                    await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(documentSpan, CancellationToken).ConfigureAwait(false);
+                var mappedDocumentSpan = await AbstractDocumentSpanEntry.TryMapAndGetFirstAsync(documentSpan, sourceText, CancellationToken).ConfigureAwait(false);
+                if (mappedDocumentSpan == null)
+                {
+                    return null;
+                }
 
                 return new DocumentSpanEntry(
-                    this, definitionBucket, documentSpan, spanKind,
-                    projectName, guid, sourceText, classifiedSpansAndHighlightSpan);
+                    this, definitionBucket, spanKind, projectName,
+                    guid, mappedDocumentSpan.Value, excerptResult);
+            }
+
+            private async Task<ExcerptResult> ExcerptAsync(SourceText sourceText, DocumentSpan documentSpan)
+            {
+                var excerptService = documentSpan.Document.Services.GetService<IDocumentExcerptService>();
+                if (excerptService != null)
+                {
+                    var result = await excerptService.TryExcerptAsync(documentSpan.Document, documentSpan.SourceSpan, ExcerptMode.SingleLine, CancellationToken).ConfigureAwait(false);
+                    if (result != null)
+                    {
+                        return result.Value;
+                    }
+                }
+
+                var classificationResult = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(documentSpan, CancellationToken).ConfigureAwait(false);
+                return new ExcerptResult(
+                    sourceText,
+                    classificationResult.HighlightSpan,
+                    classificationResult.ClassifiedSpans,
+                    documentSpan.Document,
+                    documentSpan.SourceSpan);
```

Comment:
1. why does an excerpt span not just point at a DocumentSpan?
2. why does the excerpt span need a sourcetext if it points at a document?

Reply:
DocumentSpan looks like not universal. not used anywhere else except FAR and its related code. so, decide not using. SourceText just means source text. not source text of document or anything. using sourcetext since it is cheaper than string since it support sub text/range and etc.

for number 3, you can open an issue. since I believe excertService should internally use ClassifiedSpan not the other way around.

one thing I coudl do but i didn't yet is making default Exceprt Service for normal roslyn document. so I don't need if branch here.

https://github.com/dotnet/roslyn/issues/30967

Reply:
DocumentSpan is used in a fair number of places (esp. in the 'references' space).  For example, we use it for 'Document Highlighting', and it's the standard currency for 'Definition Items/Reference Items' for find-refs.  

Give this code is underneath the Find-References umbrella, it seems pretty weird to not use it :)